source/speech/__init__.py

```diff
@@ -2210,9 +2210,13 @@ def getFormatFieldSpeech(  # noqa: C901
 			textList.append(text)
 	if formatConfig["reportSuperscriptsAndSubscripts"]:
 		textPosition=attrs.get("text-position")
+		if textPosition:
+			textPosition = textPosition.lower()
+		if textPosition not in ['super', 'sub']:
+			textPosition = None

```

Comment:
Please add a comment here about the other values it might be, and why NVDA should not report them.

Reply:
Well, that's not so clear, there are many cases.
* The following object's textInfo may report sub / super / [no field]
  * JABTextInfo (in NVDAObjects/JAB/__init__.py)
  * EditTextInfo (in sub / super / [no field])
  * ITextDocumentTextInfo (in NVDAObjects/window/edit.py:)
  * TextFrameTextInfo (in appModules/powerpnt.py:)
  * WordDocumentTextInfo (in NVDAObjects/window/winword.py: thanks to NVDAHelper code)
* The following object's textInfo may report sub / super / baseline / [no field]
  * UIATextInfo (in NVDAObjects/UIA/__init__.py:)
  * SymphonyTextInfo (in appModules/soffice.py:)
* IA2TextTextInfo (in NVDAObjects\IAccessible\__init__.py:) seems to be able to produce various values that are directly passed. I do not know this part of code tho know where it initially comes from. But the values I have already gotten are: 'auto', 'middle', '50%'.

And this list may not be exhaustive. I have made it by using:
`git grep -i "text-position"`
`git grep -i "reportSuperscriptsAndSubscripts"`
and some additional tests in IE where I had already encountered some strange values.

Anyway, my approach was to follow what was done previously, except for reporting 'baseline', thus leading to the same result if 'baseline' or nothing or any value different from 'sub' or 'super is provided.

If required, please give me further tips to have an exhausting list of where these value can come from and to know the real origin of the value provided by IA2TextTextInfo.
Thanks.

Reply:
Ok perhaps cleaning that up may best as a separate task. It would be nice if each of the providers for this info normalized the values to a some Enum value.

@michaelDCurran , do you have any suggestions for ensuring we have a "known set" of values for `textPosition`?